packages/core/src/common/document-cache.ts

```diff
@@ -127,8 +127,9 @@ export function templatePathForSynthesizedModule(path: string): string {
   return path.replace(SCRIPT_EXTENSION_REGEX, '.hbs');
 }
 
-export function synthesizedModulePathForTemplate(path: string): string {
-  return path.replace(TEMPLATE_EXTENSION_REGEX, '.ts');
+export function synthesizedModulePathForTemplate(path: string, config: GlintConfig): string {
+  let extension = config.checkStandaloneTemplates ? '.ts' : '.js';
```

Comment:
One thought: Is this naming a bit confusing? In the event that a consumer has a JS-only project but they want `checkJs: true`, they'd want the JS extension on their standalone templates but they'd also want to typecheck the file still